sklearn/datasets/tests/test_svmlight_format.py

```diff
@@ -76,24 +82,25 @@ def test_load_svmlight_file_fd():
 
     # GH20081: testing equality between path-based and
     # fd-based load_svmlight_file
-    with _path(TEST_DATA_MODULE, datafile) as data_path:
-        data_path = str(data_path)
-        X1, y1 = load_svmlight_file(data_path)
 
-        fd = os.open(data_path, os.O_RDONLY)
-        try:
-            X2, y2 = load_svmlight_file(fd)
-            assert_array_almost_equal(X1.data, X2.data)
-            assert_array_almost_equal(y1, y2)
-        finally:
-            os.close(fd)
+    data_path = resources.files(TEST_DATA_MODULE) / datafile
+    data_path = str(data_path)
+    X1, y1 = load_svmlight_file(data_path)
+
+    fd = os.open(data_path, os.O_RDONLY)
+    try:
+        X2, y2 = load_svmlight_file(fd)
+        assert_array_almost_equal(X1.data, X2.data)
+        assert_array_almost_equal(y1, y2)
+    finally:
+        os.close(fd)
 
 
 def test_load_svmlight_pathlib():
     # test loading from file descriptor
-    with _path(TEST_DATA_MODULE, datafile) as data_path:
-        X1, y1 = load_svmlight_file(str(data_path))
-        X2, y2 = load_svmlight_file(data_path)
+    data_path = _svmlight_local_test_file_path(datafile)
```

Comment:
So it seems like the `resources.as_file` which was previously used in `path` makes sense when the resource is inside a `.zip`, see https://docs.python.org/3.12/library/importlib.resources.html#importlib.resources.as_file

It seems like `sklearn/datasets/tests/test_svmlight_format.py` is the only place we care about this so I removed it.

I am not too sure we care about this too much in general since loading .so from zips is not allowed according to [this](https://docs.python.org/3/library/zipimport.html)

Reply:
Indeed, I don't think we care because it would have failed for all the other loaders. Nobody uses the `.egg` package format anymore and both `.whl` and conda packages get unzipped in the `site-packages` folder prior to importing.

If we ever have someone complaining, we can add that feature consistently for all dataset loaders in a future PR. That might be useful when using scikit-learn in a frozen executable app for instance.